### ISSUE TITLE
net-contour seems to use `--ingress-class-name` by unexpected way

### ISSUE THREAD

**Issue by nak3** [knative-sandbox/net-contour#549]
Knative deploys two Contour, `contour-external` & `contour-internal`, and uses `--ingress-class-name` for each Contour.

(non-operator)
https://github.com/knative-sandbox/net-contour/blob/076d698183c3194e02aba4e8233adf9a6d1b4fdc/config/contour/external.yaml#L2048

(operator)
https://github.com/knative-sandbox/net-contour/blob/076d698183c3194e02aba4e8233adf9a6d1b4fdc/config/contour-operator/gateway-external.yaml#L16

Because HTTPProxy(s) created by them will have `projectcontour.io/ingress.class` so we can determine the visibility by the annotation.

However, Contour expects that the ingress-class-name is used for the name of `ingressclass.networking.k8s.io`. In fact, contour operator added the validation by https://github.com/projectcontour/contour-operator/pull/388. 
And net-contour with contour operator v1.17 stops working.

**Comment by nak3** [comment]
> And net-contour with contour operator v1.17 stops working.

We can workaround the issue by creating a dummy IngressClass like:

```
apiVersion: networking.k8s.io/v1
kind: IngressClass
metadata:
  name: contour-external
  namespace: contour-external
spec:
  controller: example.com/dummy
```

However, we should ask upstream to drop the validation or change the current net-contour structure.

**Comment by nak3** [comment]
This was fixed by contour v1.18.
